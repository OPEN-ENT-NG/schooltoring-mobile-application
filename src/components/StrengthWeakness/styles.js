import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export const styleStructure = StyleSheet.create({
  header: {
    flexDirection: "row",
    alignItems: "center"
  },
  title: {
    color: COLORS.WHITE,
    fontSize: 25,
    marginLeft: 10
  },
  container: {
    paddingHorizontal: 30,
    paddingVertical: 15,
    flex: 1,
    justifyContent: "space-evenly"
  },
  list: {
    flexGrow: 1,
    flexDirection: "row",
    justifyContent: "flex-start",
    flexWrap: "wrap"
  },
  buttonView: {
    flexDirection: "row",
    justifyContent: "flex-end",
    flex: 0,
    padding: 15
  }
});

export const styleStrength = StyleSheet.create({
  header: {
    backgroundColor: COLORS.PRIMARY
  },
  items: {
    backgroundColor: COLORS.PRIMARY
  }
});

export const styleWeakness = StyleSheet.create({
  header: {
    backgroundColor: COLORS.SECONDARY
  },
  items: {
    backgroundColor: COLORS.SECONDARY
  }
});
