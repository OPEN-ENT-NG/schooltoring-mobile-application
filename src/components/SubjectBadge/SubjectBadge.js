import React, { Component } from "react";
import {
  View,
  Text} from "react-native";
import PropTypes from "prop-types";

import Touchable from "../Touchable/Touchable";

import styles from "./styles";

export default class SubjectBadge extends Component {
  render() {
    return (
      <Touchable
        onPress={() =>
          this.props.hasOwnProperty("onPress") && this.props.onPress()
        }
      >
        <View style={[styles.item, this.props.style]}>
          <Text style={styles.itemText}>{this.props.title}</Text>
        </View>
      </Touchable>
    );
  }
}

SubjectBadge.propTypes = {
  onPress: PropTypes.func,
  style: PropTypes.object,
  title: PropTypes.string.isRequired
};
