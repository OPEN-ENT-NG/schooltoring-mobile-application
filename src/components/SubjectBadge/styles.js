import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  item: {
    padding: 5,
    margin: 5,
    alignSelf: "flex-start"
  },
  itemText: {
    color: COLORS.WHITE
  }
});
