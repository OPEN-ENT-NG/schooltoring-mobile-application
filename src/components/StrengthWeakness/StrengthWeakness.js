import React, { Component } from "react";
import { View, Text, ScrollView } from "react-native";
import PropTypes from "prop-types";

import Icon from "react-native-vector-icons/MaterialIcons";
import I18n from "../../api/I18n";

import Header from "../Header/Header";
import SecondaryButton from "../SecondaryButton/SecondaryButton";
import Autocomplete from "../Autocomplete/Autocomplete";
import SubjectBadge from "../SubjectBadge/SubjectBadge";

import { styleStrength, styleWeakness, styleStructure } from "./styles";
import { COLORS } from "../../styles/common";

export default class StrengthWeakness extends Component {
  static navigationOptions = ({ navigation }) => {
    return {
      header: (
        <Header
          navigation={navigation}
          headerStyle={
            navigation.state.routeName === "Strength"
              ? styleStrength.header
              : styleWeakness.header
          }
          noBack={navigation.state.params && navigation.state.params.noBack}
        >
          <View style={styleStructure.header}>
            <Icon
              style={styleStructure.title}
              name={
                navigation.state.routeName === "Strength"
                  ? "thumb-up"
                  : "thumb-down"
              }
            />
            <Text style={styleStructure.title}>
              {I18n.t(`${navigation.state.routeName.toLowerCase()}.title`)}
            </Text>
          </View>
        </Header>
      )
    };
  };

  constructor(props) {
    super(props);

    this.state = {
      subjects:
        props.screenProps.profile[
          props.navigation.state.routeName === "Strength"
            ? "strengths"
            : "weaknesses"
        ] || []
    };
  }

  getRouteName() {
    return this.props.navigation.state.routeName;
  }

  chooseStrengthWeakness(strength, weakness) {
    return this.getRouteName() === "Strength" ? strength : weakness;
  }

  getProfileProperty() {
    return this.chooseStrengthWeakness("strengths", "weaknesses");
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <View style={styleStructure.container}>
          <Autocomplete
            placeholder="Choisissez les matières concernées"
            data={this.props.screenProps.subjects}
            getItemKey={item => item.subjectId}
            onItemPress={item => {
              let studentSubjects = [...this.state.subjects, item];
              this.setState({ subjects: studentSubjects });
            }}
            renderItem={subject => <Text>{subject.subjectLabel}</Text>}
            filterItem={(item, filter) =>
              item.subjectLabel.toLowerCase().includes(filter.toLowerCase()) &&
              !this.state.subjects.includes(item)
            }
          />

          <ScrollView
            contentContainerStyle={styleStructure.list}
            keyboardShouldPersistTaps="handled"
          >
            {this.state.subjects.map(subject => (
              <SubjectBadge
                key={subject.subjectId}
                onPress={() => {
                  let newState = this.state.subjects;
                  newState.splice(newState.indexOf(subject), 1);
                  this.setState({ subjects: newState });
                }}
                style={{
                  backgroundColor: this.chooseStrengthWeakness(
                    COLORS.PRIMARY,
                    COLORS.SECONDARY
                  )
                }}
                title={subject.subjectLabel}
              />
            ))}
          </ScrollView>
        </View>
        <View style={styleStructure.buttonView}>
          <SecondaryButton
            onPress={() => {
              this.props.screenProps.onChangeScreen(
                this.getProfileProperty(),
                []
              );
              this.props.navigation.push(
                this.chooseStrengthWeakness("Weakness", "Availability")
              );
            }}
            title={I18n.t("skip")}
          />
          <SecondaryButton
            onPress={() => {
              this.props.screenProps.onChangeScreen(
                this.getProfileProperty(),
                this.state.subjects
              );
              this.props.navigation.push(
                this.chooseStrengthWeakness("Weakness", "Availability")
              );
            }}
            title={I18n.t("next")}
          />
        </View>
      </View>
    );
  }
}
