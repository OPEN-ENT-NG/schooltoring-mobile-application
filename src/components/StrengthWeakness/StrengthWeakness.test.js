import React from "react";
import { shallow } from "enzyme";

import StrengthWeakness from "./StrengthWeakness";
import SubjectBadge from "../SubjectBadge/SubjectBadge";

import state from "../../jest/state.json";

const navigation = {
  state: {
    routeName: "Strength"
  },
  push: jest.fn(),
  pop: jest.fn()
};

const onChangeScreen = jest.fn();

const rendered = shallow(
  <StrengthWeakness
    {...state}
    onChangeScreen={onChangeScreen}
    navigation={navigation}
  />
);

const renderedWithSaveButton = shallow(
  <StrengthWeakness
    {...state}
    onChangeScreen={onChangeScreen}
    navigation={navigation}
    saveButton={true}
  />
);

describe("Render", () => {
  test("StrengthWeakness with no save button should render without crashing", () => {
    expect(rendered).toMatchSnapshot();
  });

  test("StrengthWeakness should render 3 SubjectBadge given props", () => {
    const badges = rendered.find(SubjectBadge);
    expect(badges.length).toEqual(3);
  });

  test("StrengthWeakness with no save button should render 2 secondary buttons", () => {
    expect(rendered.find("SecondaryButton").length).toEqual(2);
  });

  test("StrengthWeakness with save button should render without crashing", () => {
    expect(renderedWithSaveButton).toMatchSnapshot();
  });

  test("StrengthWeakness with save  button should render only 1 secondary button", () => {
    expect(renderedWithSaveButton.find("SecondaryButton").length).toEqual(1);
  });
});

describe("Interactions", () => {
  const skipButton = rendered.find("SecondaryButton").at(0);
  const nextButton = rendered.find("SecondaryButton").at(1);
  const saveButton = renderedWithSaveButton.find("SecondaryButton");
  test("Press 'Next' button should trigger onChangeScreen function and navigation.push function", () => {
    nextButton.props().onPress();
    expect(onChangeScreen).toHaveBeenCalled();
    expect(navigation.push).toHaveBeenCalled();
  });

  test("Press 'Skip' button should trigger onChangeScreen function and navigation.push function", () => {
    onChangeScreen.mockReset();
    navigation.push.mockReset();
    skipButton.props().onPress();

    expect(onChangeScreen).toHaveBeenCalled();
    expect(navigation.push).toHaveBeenCalled();
  });

  test("Press 'Save' button should trigger onCHangeScreen function and navigation.pop", () => {
    onChangeScreen.mockReset();
    navigation.pop.mockReset();
    saveButton.props().onPress();

    expect(onChangeScreen).toHaveBeenCalled();
    expect(navigation.pop).toHaveBeenCalled();
  });
});
