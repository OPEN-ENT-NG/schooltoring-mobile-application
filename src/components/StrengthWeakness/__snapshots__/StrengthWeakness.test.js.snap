// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render StrengthWeakness with no save button should render without crashing 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-evenly",
        "paddingHorizontal": 30,
        "paddingVertical": 15,
      }
    }
  >
    <Autocomplete
      data={
        Array [
          Object {
            "subjectCode": "ZIP-073700",
            "subjectId": "1473675-1517929209848",
            "subjectLabel": "ENSEIGNEMTS TECHNOLOGIQUES TRANSVERSAUX",
          },
          Object {
            "subjectCode": "ZIP-278300",
            "subjectId": "1473612-1517929209848",
            "subjectLabel": "ARTS APPLIQUES ET CULTURES ARTISTIQUES",
          },
          Object {
            "subjectCode": "ZIP-074400",
            "subjectId": "1473613-1517929209848",
            "subjectLabel": "ENSEIGN. GENERAUX LIES A LA SPECIALITE",
          },
          Object {
            "subjectCode": "ZIP-030602",
            "subjectId": "1473484-1517929209848",
            "subjectLabel": "ESPAGNOL LV2",
          },
          Object {
            "subjectCode": "ZIP-232200",
            "subjectId": "1473748-1517929209848",
            "subjectLabel": "CONST.STR.MAT.APPLIQ. ELECTROTECHNIQUE",
          },
          Object {
            "subjectCode": "ZIP-207900",
            "subjectId": "1473746-1517929209848",
            "subjectLabel": "CONDUITE  ET REALISATION  PROJET CLIENT",
          },
          Object {
            "subjectCode": "ZIP-207800",
            "subjectId": "1473747-1517929209848",
            "subjectLabel": "CONCEPTION DES SYSTEMES AUTOMATIQUES",
          },
          Object {
            "subjectCode": "ZIP-006600",
            "subjectId": "1473259-1517929209848",
            "subjectLabel": "ACCOMPAGNEMENT PERSONNALISE",
          },
          Object {
            "subjectCode": "ZIP-062300",
            "subjectId": "1473258-1517929209848",
            "subjectLabel": "PHYSIQUE-CHIMIE",
          },
          Object {
            "subjectCode": "ZIP-221800",
            "subjectId": "1475595-1517929209848",
            "subjectLabel": "ELECTRONIQUE ET COMMUNICATIONS",
          },
          Object {
            "subjectCode": "ZIP-065700",
            "subjectId": "1475582-1517929209848",
            "subjectLabel": "PHYSIQUE",
          },
          Object {
            "subjectCode": "ZIP-231200",
            "subjectId": "1475583-1517929209848",
            "subjectLabel": "ELECTRICITE",
          },
          Object {
            "subjectCode": "ZIP-308700",
            "subjectId": "1475581-1517929209848",
            "subjectLabel": "MESURE ET INSTRUMENTATION",
          },
          Object {
            "subjectCode": "ZIP-068100",
            "subjectId": "1475567-1517929209848",
            "subjectLabel": "SCIENCES PHYSIQUES ET CHIMIE APPLIQUEE",
          },
          Object {
            "subjectCode": "ZIP-220700",
            "subjectId": "1475566-1517929209848",
            "subjectLabel": "ESSAI DE SYSTEMES",
          },
          Object {
            "subjectCode": "ZIP-341700",
            "subjectId": "1475710-1517929209848",
            "subjectLabel": "VENTE(TP,PRESENTATION,EXPR.,COMPORTEMENT",
          },
          Object {
            "subjectCode": "ZIP-308600",
            "subjectId": "1475847-1517929209848",
            "subjectLabel": "CHIMIE-BIOCHIMIE-SCIENCES DU VIVANT",
          },
          Object {
            "subjectCode": "ZIP-062900",
            "subjectId": "1475846-1517929209848",
            "subjectLabel": "SCIENCES DE LA VIE ET DE LA TERRE",
          },
          Object {
            "subjectCode": "ZIP-003300",
            "subjectId": "1475794-1517929209848",
            "subjectLabel": "TRAVAUX INITIAT.PERSONNEL ENCADRES",
          },
          Object {
            "subjectCode": "ZIP-068700",
            "subjectId": "1475795-1517929209848",
            "subjectLabel": "SCIENCES INDUSTRIELLES DE L'INGENIEUR",
          },
          Object {
            "subjectCode": "ZIP-043700",
            "subjectId": "1475762-1517929209848",
            "subjectLabel": "HISTOIRE-GEOGRAPHIE",
          },
          Object {
            "subjectCode": "ZIP-043800",
            "subjectId": "1475236-1517929209848",
            "subjectLabel": "ENSEIGNEMENT MORAL ET CIVIQUE",
          },
          Object {
            "subjectCode": "ZIP-022700",
            "subjectId": "1475234-1517929209848",
            "subjectLabel": "FRANCAIS PHILOSOPHIE",
          },
          Object {
            "subjectCode": "ZIP-010300",
            "subjectId": "1475235-1517929209848",
            "subjectLabel": "PHILOSOPHIE",
          },
          Object {
            "subjectCode": "ZIP-023400",
            "subjectId": "1474924-1517929209848",
            "subjectLabel": "CULTURE GENERALE ET EXPRESSION",
          },
          Object {
            "subjectCode": "ZIP-020700",
            "subjectId": "1474925-1517929209848",
            "subjectLabel": "FRANCAIS",
          },
          Object {
            "subjectCode": "ZIP-005200",
            "subjectId": "1474923-1517929209848",
            "subjectLabel": "CULTURE GENERALE ET EXPRESSION",
          },
          Object {
            "subjectCode": "ZIP-030201",
            "subjectId": "1474895-1517929209848",
            "subjectLabel": "ANGLAIS LV1",
          },
          Object {
            "subjectCode": "ZIP-061300",
            "subjectId": "1474348-1517929209848",
            "subjectLabel": "MATHEMATIQUES",
          },
          Object {
            "subjectCode": "ZIP-002300",
            "subjectId": "1480227-1517929209848",
            "subjectLabel": "TRAVAUX PERSONNELS ENCADRES",
          },
          Object {
            "subjectCode": "ZIP-383400",
            "subjectId": "1480819-1517929209848",
            "subjectLabel": "ECONOMIE-DROIT",
          },
          Object {
            "subjectCode": "ZIP-380400",
            "subjectId": "1480821-1517929209848",
            "subjectLabel": "ECONOMIE ET GESTION",
          },
          Object {
            "subjectCode": "ZIP-383000",
            "subjectId": "1480820-1517929209848",
            "subjectLabel": "PRINCIPES FONDAMENT. ECONOMIE & GESTION",
          },
          Object {
            "subjectCode": "ZIP-432200",
            "subjectId": "1480822-1517929209848",
            "subjectLabel": "ECONOMIE ET GESTION DE L'ENTREPRISE",
          },
          Object {
            "subjectCode": "ZIP-373100",
            "subjectId": "1480729-1517929209848",
            "subjectLabel": "INFORMATIQUE ET SCIENCES DU NUMERIQUE",
          },
          Object {
            "subjectCode": "ZIP-231400",
            "subjectId": "1480937-1517929209848",
            "subjectLabel": "TECHNOLOGIE SCHEMA ET FABRICATION",
          },
          Object {
            "subjectCode": "ZIP-231300",
            "subjectId": "1480938-1517929209848",
            "subjectLabel": "CONSTRUCTION (ELECTROTECHNIQUE)",
          },
          Object {
            "subjectCode": "ZIP-242000",
            "subjectId": "1477513-1517929209848",
            "subjectLabel": "SYSTEMES D'INFORMATION ET NUMERIQUE",
          },
          Object {
            "subjectCode": "ZIP-207300",
            "subjectId": "1477371-1517929209848",
            "subjectLabel": "GENIE MECANIQUE",
          },
          Object {
            "subjectCode": "ZIP-501100",
            "subjectId": "1477152-1517929209848",
            "subjectLabel": "ENSEIGNEMENT TECHNOLOGIQUE EN LV1",
          },
          Object {
            "subjectCode": "ZIP-181700",
            "subjectId": "1477852-1517929209848",
            "subjectLabel": "ENERGIES ET ENVIRONNEMENT",
          },
          Object {
            "subjectCode": "ZIP-073600",
            "subjectId": "1477779-1517929209848",
            "subjectLabel": "INNOVATION TECHNOLOGIQUE ET ECO-CONCEPT.",
          },
          Object {
            "subjectCode": "ZIP-240500",
            "subjectId": "1476487-1517929209848",
            "subjectLabel": "INFORMATIQUE",
          },
          Object {
            "subjectCode": "ZIP-062400",
            "subjectId": "1476436-1517929209848",
            "subjectLabel": "SCIENCES PHYSIQUES",
          },
          Object {
            "subjectCode": "ZIP-230700",
            "subjectId": "1476027-1517929209848",
            "subjectLabel": "ELECTROTECHNIQUE",
          },
          Object {
            "subjectCode": "ZIP-100100",
            "subjectId": "1476565-1517929209848",
            "subjectLabel": "EDUCATION PHYSIQUE ET SPORTIVE",
          },
          Object {
            "subjectCode": "ZIP-312800",
            "subjectId": "1479468-1517929209848",
            "subjectLabel": "PREVENTION-SANTE-ENVIRONNEMENT",
          },
          Object {
            "subjectCode": "ZIP-212000",
            "subjectId": "1479278-1517929209848",
            "subjectLabel": "MECANIQUE ET VIBRATIONS",
          },
          Object {
            "subjectCode": "ZIP-181400",
            "subjectId": "1479277-1517929209848",
            "subjectLabel": "THERMODYNAMIQUE ET THERMIQUE",
          },
          Object {
            "subjectCode": "ZIP-061900",
            "subjectId": "1479276-1517929209848",
            "subjectLabel": "OPTIQUE OPTOELECTRONIQUE ET IMAGERIE",
          },
          Object {
            "subjectCode": "ZIP-030102",
            "subjectId": "1480091-1517929209848",
            "subjectLabel": "ALLEMAND LV2",
          },
          Object {
            "subjectCode": "ZIP-063300",
            "subjectId": "1479696-1517929209848",
            "subjectLabel": "CHIMIE",
          },
          Object {
            "subjectCode": "ZIP-222100",
            "subjectId": "1478196-1517929209848",
            "subjectLabel": "GENIE ELECTRIQUE",
          },
          Object {
            "subjectCode": "ZIP-232100",
            "subjectId": "1478613-1517929209848",
            "subjectLabel": "SCIENCES APPLIQUEES ELECTRIQ ELECTRONIQ",
          },
          Object {
            "subjectCode": "ZIP-373700",
            "subjectId": "1478714-1517929209848",
            "subjectLabel": "INFORMATIQUE ET CREATION NUMERIQUE",
          },
          Object {
            "subjectCode": "ZIP-069700",
            "subjectId": "1478744-1517929209848",
            "subjectLabel": "SCIENCES PHYSIQUES ET CHIMIQUES EN LABO.",
          },
        ]
      }
      filterItem={[Function]}
      getItemKey={[Function]}
      onItemPress={[Function]}
      placeholder="Choisissez les matières concernées"
      renderItem={[Function]}
      underlineColor="#17b6b0"
    />
    <ScrollView
      contentContainerStyle={
        Object {
          "flexDirection": "row",
          "flexGrow": 1,
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        }
      }
      keyboardShouldPersistTaps="handled"
    >
      <SubjectBadge
        key="0"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#17b6b0",
          }
        }
        title="SCIENCES PHYSIQUES ET CHIMIE APPLIQUEE"
      />
      <SubjectBadge
        key="1"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#17b6b0",
          }
        }
        title="SCIENCES INDUSTRIELLES DE L'INGENIEUR"
      />
      <SubjectBadge
        key="2"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#17b6b0",
          }
        }
        title="MATHEMATIQUES"
      />
    </ScrollView>
  </Component>
  <Component
    style={
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "padding": 15,
      }
    }
  >
    <SecondaryButton
      onPress={[Function]}
      title="PASSER"
    />
    <SecondaryButton
      onPress={[Function]}
      title="SUIVANT"
    />
  </Component>
</Component>
`;

exports[`Render StrengthWeakness with save button should render without crashing 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-evenly",
        "paddingHorizontal": 30,
        "paddingVertical": 15,
      }
    }
  >
    <Autocomplete
      data={
        Array [
          Object {
            "subjectCode": "ZIP-073700",
            "subjectId": "1473675-1517929209848",
            "subjectLabel": "ENSEIGNEMTS TECHNOLOGIQUES TRANSVERSAUX",
          },
          Object {
            "subjectCode": "ZIP-278300",
            "subjectId": "1473612-1517929209848",
            "subjectLabel": "ARTS APPLIQUES ET CULTURES ARTISTIQUES",
          },
          Object {
            "subjectCode": "ZIP-074400",
            "subjectId": "1473613-1517929209848",
            "subjectLabel": "ENSEIGN. GENERAUX LIES A LA SPECIALITE",
          },
          Object {
            "subjectCode": "ZIP-030602",
            "subjectId": "1473484-1517929209848",
            "subjectLabel": "ESPAGNOL LV2",
          },
          Object {
            "subjectCode": "ZIP-232200",
            "subjectId": "1473748-1517929209848",
            "subjectLabel": "CONST.STR.MAT.APPLIQ. ELECTROTECHNIQUE",
          },
          Object {
            "subjectCode": "ZIP-207900",
            "subjectId": "1473746-1517929209848",
            "subjectLabel": "CONDUITE  ET REALISATION  PROJET CLIENT",
          },
          Object {
            "subjectCode": "ZIP-207800",
            "subjectId": "1473747-1517929209848",
            "subjectLabel": "CONCEPTION DES SYSTEMES AUTOMATIQUES",
          },
          Object {
            "subjectCode": "ZIP-006600",
            "subjectId": "1473259-1517929209848",
            "subjectLabel": "ACCOMPAGNEMENT PERSONNALISE",
          },
          Object {
            "subjectCode": "ZIP-062300",
            "subjectId": "1473258-1517929209848",
            "subjectLabel": "PHYSIQUE-CHIMIE",
          },
          Object {
            "subjectCode": "ZIP-221800",
            "subjectId": "1475595-1517929209848",
            "subjectLabel": "ELECTRONIQUE ET COMMUNICATIONS",
          },
          Object {
            "subjectCode": "ZIP-065700",
            "subjectId": "1475582-1517929209848",
            "subjectLabel": "PHYSIQUE",
          },
          Object {
            "subjectCode": "ZIP-231200",
            "subjectId": "1475583-1517929209848",
            "subjectLabel": "ELECTRICITE",
          },
          Object {
            "subjectCode": "ZIP-308700",
            "subjectId": "1475581-1517929209848",
            "subjectLabel": "MESURE ET INSTRUMENTATION",
          },
          Object {
            "subjectCode": "ZIP-068100",
            "subjectId": "1475567-1517929209848",
            "subjectLabel": "SCIENCES PHYSIQUES ET CHIMIE APPLIQUEE",
          },
          Object {
            "subjectCode": "ZIP-220700",
            "subjectId": "1475566-1517929209848",
            "subjectLabel": "ESSAI DE SYSTEMES",
          },
          Object {
            "subjectCode": "ZIP-341700",
            "subjectId": "1475710-1517929209848",
            "subjectLabel": "VENTE(TP,PRESENTATION,EXPR.,COMPORTEMENT",
          },
          Object {
            "subjectCode": "ZIP-308600",
            "subjectId": "1475847-1517929209848",
            "subjectLabel": "CHIMIE-BIOCHIMIE-SCIENCES DU VIVANT",
          },
          Object {
            "subjectCode": "ZIP-062900",
            "subjectId": "1475846-1517929209848",
            "subjectLabel": "SCIENCES DE LA VIE ET DE LA TERRE",
          },
          Object {
            "subjectCode": "ZIP-003300",
            "subjectId": "1475794-1517929209848",
            "subjectLabel": "TRAVAUX INITIAT.PERSONNEL ENCADRES",
          },
          Object {
            "subjectCode": "ZIP-068700",
            "subjectId": "1475795-1517929209848",
            "subjectLabel": "SCIENCES INDUSTRIELLES DE L'INGENIEUR",
          },
          Object {
            "subjectCode": "ZIP-043700",
            "subjectId": "1475762-1517929209848",
            "subjectLabel": "HISTOIRE-GEOGRAPHIE",
          },
          Object {
            "subjectCode": "ZIP-043800",
            "subjectId": "1475236-1517929209848",
            "subjectLabel": "ENSEIGNEMENT MORAL ET CIVIQUE",
          },
          Object {
            "subjectCode": "ZIP-022700",
            "subjectId": "1475234-1517929209848",
            "subjectLabel": "FRANCAIS PHILOSOPHIE",
          },
          Object {
            "subjectCode": "ZIP-010300",
            "subjectId": "1475235-1517929209848",
            "subjectLabel": "PHILOSOPHIE",
          },
          Object {
            "subjectCode": "ZIP-023400",
            "subjectId": "1474924-1517929209848",
            "subjectLabel": "CULTURE GENERALE ET EXPRESSION",
          },
          Object {
            "subjectCode": "ZIP-020700",
            "subjectId": "1474925-1517929209848",
            "subjectLabel": "FRANCAIS",
          },
          Object {
            "subjectCode": "ZIP-005200",
            "subjectId": "1474923-1517929209848",
            "subjectLabel": "CULTURE GENERALE ET EXPRESSION",
          },
          Object {
            "subjectCode": "ZIP-030201",
            "subjectId": "1474895-1517929209848",
            "subjectLabel": "ANGLAIS LV1",
          },
          Object {
            "subjectCode": "ZIP-061300",
            "subjectId": "1474348-1517929209848",
            "subjectLabel": "MATHEMATIQUES",
          },
          Object {
            "subjectCode": "ZIP-002300",
            "subjectId": "1480227-1517929209848",
            "subjectLabel": "TRAVAUX PERSONNELS ENCADRES",
          },
          Object {
            "subjectCode": "ZIP-383400",
            "subjectId": "1480819-1517929209848",
            "subjectLabel": "ECONOMIE-DROIT",
          },
          Object {
            "subjectCode": "ZIP-380400",
            "subjectId": "1480821-1517929209848",
            "subjectLabel": "ECONOMIE ET GESTION",
          },
          Object {
            "subjectCode": "ZIP-383000",
            "subjectId": "1480820-1517929209848",
            "subjectLabel": "PRINCIPES FONDAMENT. ECONOMIE & GESTION",
          },
          Object {
            "subjectCode": "ZIP-432200",
            "subjectId": "1480822-1517929209848",
            "subjectLabel": "ECONOMIE ET GESTION DE L'ENTREPRISE",
          },
          Object {
            "subjectCode": "ZIP-373100",
            "subjectId": "1480729-1517929209848",
            "subjectLabel": "INFORMATIQUE ET SCIENCES DU NUMERIQUE",
          },
          Object {
            "subjectCode": "ZIP-231400",
            "subjectId": "1480937-1517929209848",
            "subjectLabel": "TECHNOLOGIE SCHEMA ET FABRICATION",
          },
          Object {
            "subjectCode": "ZIP-231300",
            "subjectId": "1480938-1517929209848",
            "subjectLabel": "CONSTRUCTION (ELECTROTECHNIQUE)",
          },
          Object {
            "subjectCode": "ZIP-242000",
            "subjectId": "1477513-1517929209848",
            "subjectLabel": "SYSTEMES D'INFORMATION ET NUMERIQUE",
          },
          Object {
            "subjectCode": "ZIP-207300",
            "subjectId": "1477371-1517929209848",
            "subjectLabel": "GENIE MECANIQUE",
          },
          Object {
            "subjectCode": "ZIP-501100",
            "subjectId": "1477152-1517929209848",
            "subjectLabel": "ENSEIGNEMENT TECHNOLOGIQUE EN LV1",
          },
          Object {
            "subjectCode": "ZIP-181700",
            "subjectId": "1477852-1517929209848",
            "subjectLabel": "ENERGIES ET ENVIRONNEMENT",
          },
          Object {
            "subjectCode": "ZIP-073600",
            "subjectId": "1477779-1517929209848",
            "subjectLabel": "INNOVATION TECHNOLOGIQUE ET ECO-CONCEPT.",
          },
          Object {
            "subjectCode": "ZIP-240500",
            "subjectId": "1476487-1517929209848",
            "subjectLabel": "INFORMATIQUE",
          },
          Object {
            "subjectCode": "ZIP-062400",
            "subjectId": "1476436-1517929209848",
            "subjectLabel": "SCIENCES PHYSIQUES",
          },
          Object {
            "subjectCode": "ZIP-230700",
            "subjectId": "1476027-1517929209848",
            "subjectLabel": "ELECTROTECHNIQUE",
          },
          Object {
            "subjectCode": "ZIP-100100",
            "subjectId": "1476565-1517929209848",
            "subjectLabel": "EDUCATION PHYSIQUE ET SPORTIVE",
          },
          Object {
            "subjectCode": "ZIP-312800",
            "subjectId": "1479468-1517929209848",
            "subjectLabel": "PREVENTION-SANTE-ENVIRONNEMENT",
          },
          Object {
            "subjectCode": "ZIP-212000",
            "subjectId": "1479278-1517929209848",
            "subjectLabel": "MECANIQUE ET VIBRATIONS",
          },
          Object {
            "subjectCode": "ZIP-181400",
            "subjectId": "1479277-1517929209848",
            "subjectLabel": "THERMODYNAMIQUE ET THERMIQUE",
          },
          Object {
            "subjectCode": "ZIP-061900",
            "subjectId": "1479276-1517929209848",
            "subjectLabel": "OPTIQUE OPTOELECTRONIQUE ET IMAGERIE",
          },
          Object {
            "subjectCode": "ZIP-030102",
            "subjectId": "1480091-1517929209848",
            "subjectLabel": "ALLEMAND LV2",
          },
          Object {
            "subjectCode": "ZIP-063300",
            "subjectId": "1479696-1517929209848",
            "subjectLabel": "CHIMIE",
          },
          Object {
            "subjectCode": "ZIP-222100",
            "subjectId": "1478196-1517929209848",
            "subjectLabel": "GENIE ELECTRIQUE",
          },
          Object {
            "subjectCode": "ZIP-232100",
            "subjectId": "1478613-1517929209848",
            "subjectLabel": "SCIENCES APPLIQUEES ELECTRIQ ELECTRONIQ",
          },
          Object {
            "subjectCode": "ZIP-373700",
            "subjectId": "1478714-1517929209848",
            "subjectLabel": "INFORMATIQUE ET CREATION NUMERIQUE",
          },
          Object {
            "subjectCode": "ZIP-069700",
            "subjectId": "1478744-1517929209848",
            "subjectLabel": "SCIENCES PHYSIQUES ET CHIMIQUES EN LABO.",
          },
        ]
      }
      filterItem={[Function]}
      getItemKey={[Function]}
      onItemPress={[Function]}
      placeholder="Choisissez les matières concernées"
      renderItem={[Function]}
      underlineColor="#17b6b0"
    />
    <ScrollView
      contentContainerStyle={
        Object {
          "flexDirection": "row",
          "flexGrow": 1,
          "flexWrap": "wrap",
          "justifyContent": "flex-start",
        }
      }
      keyboardShouldPersistTaps="handled"
    >
      <SubjectBadge
        key="0"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#17b6b0",
          }
        }
        title="SCIENCES PHYSIQUES ET CHIMIE APPLIQUEE"
      />
      <SubjectBadge
        key="1"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#17b6b0",
          }
        }
        title="SCIENCES INDUSTRIELLES DE L'INGENIEUR"
      />
      <SubjectBadge
        key="2"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#17b6b0",
          }
        }
        title="MATHEMATIQUES"
      />
    </ScrollView>
  </Component>
  <Component
    style={
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "padding": 15,
      }
    }
  >
    <SecondaryButton
      onPress={[Function]}
      title="ENREGISTRER"
    />
  </Component>
</Component>
`;
